### ISSUE TITLE
validate_presence should work for date fields

### ISSUE THREAD

**Issue by robj** [sxross/MotionModel#91]
It is not immediately apparent from docs it only supports Numeric/String

my custom validator for the interim:

```
def validate_date_presence(field, value, setting)
  if(value.is_a?(Time || Date))
    return true
  else
    return false
  end
end
```

**Comment by digitalfx** [comment]
This has just had me going round in circle too until I decided to remove my validations. I tried validating the presence of a date and an array, both of which failed even though they were present. I was going to call the following line on each field change in my ViewController to disable the save button until all the required fields were filled in.
`saveButton.enabled = myNewObject.valid?`
This seems like valid logic to me?

**Comment by sxross** [comment]
I'm not sure this will work. The object does not have any data until you put it there. If you aren't doing that as the user enters it, you can't validate against it. Further, once you assign to a model field of type `date` or type `array`, MotionModel casts the value to it's Ruby type. So, presence validates against `nil` or `length == 0`. The latter case may work for `array` but not for `date`. This may not be a model issue.

**Comment by digitalfx** [comment]
Ok so in which case, if I'm migrating some code from CDQ, I'd implement `datetime :mydate, optional: false` with a format validation as shown above? And for an array, I probably need to make a validation with a block of some sort right?

**Comment by sxross** [comment]
I'm not sure about the `optional:` part. I think I'd go with `default:` if you want to set it to something. If you are migrating from CDQ will you have a `string` or an `NSDate`. MotionModel will recognize `NSDate` and copy it over. If it's a string, it will try to use a `NSDataDetector` to parse it.

For the array, `presence:` looks at the length. Once you have assigned the array to your model, it will know the length and be able to validate. So you could as easily say: `length: [4..12]` or something like that.

Remember, validations are not necessarily for doing on-the-fly form validations -- more so to ensure data integrity. There's no reason not to use them for input validations, but don't forget to roll back your transaction before moving on if the user cancels.

The main place I'm looking is motion/validatable.rb. It's pretty straightforward code.

**Comment by digitalfx** [comment]
Yep I've not an NSDate, so everything is easy there. Thanks for the help, this makes more sense now :)
